### PR TITLE
Fix broken text appearances when style resource names use periods

### LINKED ISSUE #182: Support for text appearance
`android:textAppearance` in XML and `setTextAppearance()` in code are both not recognized in a paparazzi test.

Would be nice to have this working!

### PR DESCRIPTION
**Pull request by jrodbx**
Closes #182

### CODE REVIEW

File: paparazzi-gradle-plugin/src/test/projects/text-appearances-code/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <!-- Empty style for when you don't want a parent. -->
+  <style name="None" />
+
+  <style name="TextAppearance.Cash.Title" parent="None">
```
Comment (jamesmosley55) [inline paparazzi-gradle-plugin/src/test/projects/text-appearances-code/src/main/res/values/styles.xml:6]: Should this have a Cash name? or just a generic name?

File: paparazzi-gradle-plugin/src/test/projects/text-appearances-xml/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <!-- Empty style for when you don't want a parent. -->
+  <style name="None" />
+
+  <style name="TextAppearance.Cash.Title" parent="None">
```
Comment (jamesmosley55) [inline paparazzi-gradle-plugin/src/test/projects/text-appearances-xml/src/main/res/values/styles.xml:6]: ditto

File: paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt [hunk 3]
```diff
@@ -97,7 +98,15 @@ internal class PaparazziCallback(
 
   override fun resolveResourceId(id: Int): ResourceReference? = projectResources[id]
 
-  override fun getOrGenerateResourceId(resource: ResourceReference): Int = resources[resource] ?: 0
+  override fun getOrGenerateResourceId(resource: ResourceReference): Int {
+    // Workaround: We load our resource map from fields in R.class, which are named using Java
+    // class conventions.  Therefore, we need to similarly transform style naming conventions
+    // that contain periods (e.g., Widget.AppCompat.TextView) to avoid false lookup misses.
```
Comment (jamesmosley55) [inline paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt:104]: so Widget.AppCompat.Textview becomes `Widget_AppCompat_TextView`?

Seems odd, but if its needed then 🤷

**Review by jamesmosley55 — APPROVED** [review]

File: paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt (hunk 3 above)
Comment (jrodbx) [reply paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt:104]: yea, cuz we use R.class to bootstrap the resource map.  Eventually, we'll prob use another seed, which means this likely would go away/be replaced.

File: paparazzi-gradle-plugin/src/test/projects/text-appearances-code/src/main/res/values/styles.xml (hunk 1 above)
Comment (jrodbx) [reply paparazzi-gradle-plugin/src/test/projects/text-appearances-code/src/main/res/values/styles.xml:6]: can replace

File: paparazzi-gradle-plugin/src/test/projects/text-appearances-xml/src/main/res/values/styles.xml (hunk 2 above)
Comment (jrodbx) [reply paparazzi-gradle-plugin/src/test/projects/text-appearances-xml/src/main/res/values/styles.xml:6]: ditto

Comment (jrodbx) [reply paparazzi-gradle-plugin/src/test/projects/text-appearances-xml/src/main/res/values/styles.xml]: ditto

**Pull request merged**